components/authz-service/server/v2/system_integration_test.go

```diff
@@ -57,6 +59,103 @@ func TestIntegrationSystemPolicies(t *testing.T) {
 	}
 }
 
+// bug: this test passes contingent on the above test being run
+// issue seems to be related to context metadata
+func TestIntegrationValidateProjectAssignment(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	ts := setupWithOPAV2p1(t)
+	defer ts.Shutdown(t, ctx)
+	cl := ts.Authz
+
+	user := "user:local:alice"
+	authorizedProjectId := "authorized-project"
+	unauthorizedProjectId := "project-not-authorized"
+
+	_, err := ts.Projects.CreateProject(ctx, &api_v2.CreateProjectReq{
+		Id:   authorizedProjectId,
+		Name: "Project Authorized",
+	})
+	require.NoError(t, err)
+
+	_, err = ts.Projects.CreateProject(ctx, &api_v2.CreateProjectReq{
+		Id:   unauthorizedProjectId,
+		Name: "Project Unauthorized",
+	})
+	require.NoError(t, err)
+
+	statement := api_v2.Statement{
+		Effect:    api_v2.Statement_ALLOW,
+		Resources: []string{"*"},
+		Actions:   []string{"iam:projects:assign"},
+		Projects:  []string{authorizedProjectId},
+	}
+	req := api_v2.CreatePolicyReq{
+		Id:         "policy-2",
+		Name:       "my favorite policy",
+		Members:    []string{user},
+		Statements: []*api_v2.Statement{&statement},
+	}
+	_, err = ts.Policy.CreatePolicy(ctx, &req)
+	require.NoError(t, err)
+
+	// force sync refresh
```

Comment:
Could we be more specific? What are we refreshing? Right now the comment does not really add any value.